app/src/main/java/org/gdg/frisbee/android/widget/ChapterSelectDialog.java

```diff
@@ -0,0 +1,176 @@
+package org.gdg.frisbee.android.widget;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatDialogFragment;
+import android.support.v7.widget.SearchView;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Adapter;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Filter;
+import android.widget.ListView;
+
+import org.gdg.frisbee.android.R;
+import org.gdg.frisbee.android.api.model.Chapter;
+import org.gdg.frisbee.android.view.FilterListView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class ChapterSelectDialog extends AppCompatDialogFragment
+    implements AdapterView.OnItemClickListener {
+
+    private static final String EXTRA_CHAPTERS = "EXTRA_CHAPTERS";
+    private static final String EXTRA_SELECTED_CHAPTER = "EXTRA_SELECTED_CHAPTER";
+
+    @BindView(R.id.filter) SearchView cityNameSearchView;
+    @BindView(android.R.id.list) FilterListView listView;
+
+    private List<Chapter> chapters;
+
+    private Listener listener = Listener.EMPTY;
+    private Chapter selectedChapter;
+
+    public static ChapterSelectDialog newInstance(ArrayList<Chapter> chapters,
+                                                  @Nullable Chapter selectedChapter) {
+        ChapterSelectDialog fragment = new ChapterSelectDialog();
+        Bundle args = new Bundle(2);
+        args.putParcelableArrayList(EXTRA_CHAPTERS, chapters);
+        args.putParcelable(EXTRA_SELECTED_CHAPTER, selectedChapter);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        chapters = getArguments().getParcelableArrayList(EXTRA_CHAPTERS);
```

Comment:
Then, do I have to add the logic to get them from network if they are not available in the cache? Is that even be possible? 
If we are going to use it in Settings, it will definitely be possible.